### PR TITLE
Add explanation of DMOJ contest formats

### PR DESCRIPTION
**Pull request by kiritofeng**

### CODE REVIEW

File: docs/site/contest_formats.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+## Contest Formats
+
+The DMOJ ships with 4 contest formats by default: Default, Codechef IOI Ranklist (henceforth shortened to simply IOI), ECOO, and AtCoder.
+
+### Default Contest Format
+
+The `Default` contest format is what all contests ran on pre-April 23, 2019.
+
+The score is the sum of the maximum scores on each problem, and ties are broken by adding the time of the last submission to each problem with a non-zero maximum score.
```
Comment (Xyene) [inline docs/site/contest_formats.md]: Emphasize the fact that _any_ submission bumps up the time a bit more.

File: docs/site/contest_formats.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+This field defaults to `10`.
+
+The `time_bonus` field awards a bonus for solving problems faster.
+The field is in minutes, and for each such interval of time before the contest ends, any submission with a non-zero score will have a bonus added. This field defaults to `5`.  
+Note that specifying `0` will disable this bonus.  
+For example, say a submission with a score of `50/100` is submitted 23 minutes before the contest ends. $\left\lfloor\frac{23}{5}\right\rfloor = 4$, so 4 bonus points are awarded, giving a total score of $50 + 4 = 54$ for that problem.
+
+## AtCoder Contest Format
+
+The `AtCoder` contest format is based on the contest format used by AtCoder. As such, your score equals the sum of your highest-scoring submission on each problem, and ties are broken based on your last score-changing submissions plus your penalty.
```
Comment (xiaowuc1) [inline docs/site/contest_formats.md]: based on the time of your last score-changing submission

File: docs/site/contest_formats.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+The penalty is specified by the `penalty` field, and defaults to 5 minutes.
+The penalty is equal to the total number of incorrect submissions on each solved problem, multiplied by the specified value, in minutes, and is added to the cumulative time.
```
Comment (xiaowuc1) [inline docs/site/contest_formats.md]: total number of incorrect submissions prior to the highest scoring submission on each solved problem?

File: docs/site/contest_formats.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+This field defaults to `10`.
+
+The `time_bonus` field awards a bonus for solving problems faster.
+The field is in minutes, and for each such interval of time before the contest ends, any submission with a non-zero score will have a bonus added. This field defaults to `5`.  
+Note that specifying `0` will disable this bonus.  
+For example, say a submission with a score of `50/100` is submitted 23 minutes before the contest ends. $\left\lfloor\frac{23}{5}\right\rfloor = 4$, so 4 bonus points are awarded, giving a total score of $50 + 4 = 54$ for that problem.
+
+## AtCoder Contest Format
+
+The `AtCoder` contest format is based on the contest format used by AtCoder. As such, your score equals the sum of your highest-scoring submission on each problem, and ties are broken based on your last score-changing submission plus your penalty.
```
Comment (quantum5) [inline docs/site/contest_formats.md]: I think @xiaowuc1 asked for you to say *the time of* your last score-changing submission.

**Pull request merged**